### ISSUE TITLE
[enh] Support for group attribute access in expressions

### ISSUE THREAD

**Issue by nyurik** [vega/vega#990]
Please allow expression access to the group's attributes. The most useful would be to get the width/height of the immediate and parent groups.

```js
// This gets the width of the current (parent) container
// either the enclosing group, or the plot area (not the same as the graph's "width")
getAttr('width')
// get the width of grand-parent container,
// or if current group is first level, gets the width of the whole graph's plot area
getAttr('width', -1)
// get the width of the plot area
getAttr('width', null)
// get the width of the named group
getAttr('width', 'groupname')
```
Ideally, all other attributes of a group, including x,y positioning and even fill color should be available via this method.

**Comment by jheer** [comment]
Within an encoding signal expression, parent groups are already accessible via `item.mark.group`.

Alternatively, group fields can be accessed within an encoder using the `field` directive. For more, see https://vega.github.io/vega/docs/marks/#fieldref

**Comment by nyurik** [comment]
@jheer I wasn't talking about the parent's **data**, but rather parent's **attributes** like width/height, and possibly more complex (reflection-style) things strikeColor.  In a mark, I often want to get the width/height of the current group, or the parent group.  I know that x,y are relative to group's x,y.

**Comment by jheer** [comment]
So was I. The `parent` identifier accesses the parent group's datum. What I described above accesses the group scenegraph item, which houses the visual properties.

**Comment by nyurik** [comment]
Thanks @jheer, I wasn't aware of the "item" keyword. Is it documented anywhere? I only found "datum" and "parent", but "item" is not mentioned. What properties are expected on it?  (I see them in the debugger, but I doubt most users will find them). BTW, my only way to figure things out is to create a "valid but crashing expression" - e.g.  `"parent.blah.blah"` -- this gets processed by initial vega parser, but throws an exception during execution. Catching that expression in debugger allows me to examine the context of that expression. Sadly, this is the only way I found to debug anything. I hope the debugger will get migrated to Vega 3, and we can easily integrate it into end user env. Thanks!
